build_ngtf.py

```diff
@@ -291,9 +291,24 @@ def main():
             os.chdir(pwd_now)
 
             # Next install the tensorflow python packge
-            command_executor(
-                ["pip", "install", "-U", "tensorflow==" + tf_version])
-            cxx_abi = get_tf_cxxabi()
+            if ('1.15' not in tf_version):
```

Comment:
Not sure about this change. In the option `use prebuilt tf` we should pip install tf 1.15 and built. 
But it seems we are building tf from source first.

Reply:
The prebuilt TF v1.15.0 from `Google` is badly broken and we can use it with either GCC `4.8` or newer to build the bridge. This is what this whole PR is about.
Hopefully for `TF v2.1` and after there is only one GCC we have to deal with which is `7.3` and we should be good to use upstream wheels again.

Reply:
You mean, we cant use it to build with gcc 4.8 or higher ?

Reply:
Yes we can't use it on either `4.8` or newer to build bridge so we have to rebuilt it with right flags first.

Reply:
Wondering if we should error out in that case rather than building from source. We have few other options:
1. vanilla build : builds tf fom source + builds all our cpp applications
2. --use_tf_from_location : re-uses the .whl and certain tf libraries to builds bridge -> builds our cpp applications
3. --use_prebuilt_tf : pip installs tf -> cannot build our cpp applications using this method

Reply:
If we error out then our pipeline that tests this will not work. Cant think of any other place where this would be a problem.

Reply:
Options `1` and `2` are still working and it's only the option `3` that's problematic.

I'm Ok with erring out but then do we ask user to build TF themselves and pass it down to us so we can use their wheel?
If we don't have control over how they build the wheel, we could run into the same issue with their wheel as well.

Now as far as pipeline test is concerned we can leave it the way it is I guess.

Reply:
We can error out and ask the user to build using options 1 or 2 temporarily. 
In either case [erroring out or building from src and then using the .whl], It would be good to document the reason in a comment in the else section.

Reply:
Lets error out in this case. We are still testing the "pip install" pipeline, but for actual usage since we do not support "pip install tensorflow" we should error out. For example, this TF building has bazel dependency etc. Someone expecting to do "pip install tensorflow" may not have bazel dependency etc.

Reply:
Sounds good. I fixed it.

Reply:
Notes for myself:
the way its set up right now is that if an user tries to run it it will fail as artifacts_dir/tensorflow/whl will not exist

But for buildkite it will exist as we are copying in the whl in the BK yaml. So BK will proceed

Reply:
💯